specification/cognitiveservices/ContentSafety/models.tsp

```diff
@@ -180,3 +258,346 @@ model RemoveTextBlocklistItemsOptions {
+  @maxLength(1000)
+  text?: string;
+
+  @doc("The categories will be analyzed. If they are not assigned, a default set of analysis results for the categories will be returned.")
+  categories?: ImageWithTextCategory[];
+
+  @doc("When set to true, our service will perform OCR and concatenate the recognized text with input text before analyzing. We will recognize at most 256 characters (unicode code points) from input image. The others will be truncated.")
+  enableOcr: boolean;
+}
+
+@added(Versions.v2023_10_30_Preview)
+@doc("ImageWithText analysis result.")
+model ImageWithTextCategoriesAnalysis {
+  @doc("The imageWithtext analysis category.")
+  category: ImageWithTextCategory;
+
+  @doc("The higher the severity of input content, the larger this value is. The values could be: 0,2,4,6.")
+  severity?: int32;
+}
+
+@added(Versions.v2023_10_30_Preview)
+@doc("The analysis response of the image with text.")
+model AnalyzeImageWithTextResult {
+  @doc("Analysis result for categories.")
+  categoriesAnalysis: ImageWithTextCategoriesAnalysis[];
+}
+
+@added(Versions.v2023_11_30_Preview)
+@doc("The text to detect ungroundedness.")
+model DetectUngroundednessInputContent {
+  @doc("Identity of the text.")
+  id: string;
+
+  @doc("Text for analysis.")
+  text: string;
+}
+
+@added(Versions.v2023_11_30_Preview)
+@doc("User input for ungroudedness detection.")
+model DetectUngroundednessInput {
+  @doc("This is optional and only used for QnA or chat task.")
+  query?: string;
+
+  @doc("Content for analysis.")
+  contents: DetectUngroundednessInputContent[];
+
+  @doc("Source information that serves as grounding source.")
+  groundingSource: string[];
+}
+
+@added(Versions.v2023_11_30_Preview)
+@doc("Connection information for GPT resource.")
+model DetectUngroundednessGptxOptions {
+  @doc("Endpoint of GPT resource.")
+  azureOpenAIEndpoint: string;
+
+  @doc("Deployment id of GPT resource.")
+  deploymentId: string;
+}
+
+@added(Versions.v2023_11_30_Preview)
+@doc("The request of ungroudedness detection.")
+model DetectUngroundednessOptions {
+  @doc("Domain that the current request belongs to.")
+  domain: string;
+
+  @doc("Type of work that the customer want the API to execute.")
+  task: string;
+
+  @doc("The input to be examined for detection.")
+  input: DetectUngroundednessInput;
+
+  @doc("Options to set GPT resources for BYO scenario.")
+  gptxOptions?: DetectUngroundednessGptxOptions;
+}
+
+@added(Versions.v2023_11_30_Preview)
+@doc("The detection result of ungroudedness.")
+model TextUngroundednessDetection {
+  @doc("Identity of content.")
+  id: string;
+
+  @doc("Detection result for ungroudedness.")
+  isUngrounded: boolean;
+
+  @doc("The confidence store of the model in the analysis results.")
+  score: float64;
+
+  @doc("Ungroundedness rate of content.")
+  ungroundednessRate: float64;
+
+  @doc("The detailed information about a text identified as ungroundness.")
+  textSpan: UngroundednessTextSpan[];
```

Comment:
textSpan is not a good name for me, consider "details"? Besides, since this is an array, it should be plural